src/domain/event/utils.ts

```diff
@@ -511,6 +518,198 @@ export const addImageSchema = Yup.object().shape(
+    case 'Tämä kenttä ei voi olla tyhjä.':
+      return t(`event.serverError.required`);
+    case 'Tämän luvun on oltava vähintään 0.':
+      return t(`event.serverError.min0`);
+    default:
+      return errorStr;
+  }
+};
+
+const getLocalizedServerError = ({
+  error,
+  key,
+  t,
+  eventType,
+}: {
+  error: LEServerError;
+  eventType: string;
+  key: string;
+  t: TFunction;
+}): ServerErrorItem[] => {
+  /* istanbul ignore else */
+  if (typeof error === 'object') {
+    return Object.entries(error).reduce(
+      (previous: ServerErrorItem[], [lang, e]) => [
+        ...previous,
+        {
+          label: getEventServerErrorLabel({
+            key,
+            eventType,
+            langText: lowerCaseFirstLetter(t(`form.inLanguage.${lang}`)),
+            t,
+          }),
+          message: getEventServerErrorMessage({ error: [e] as string[], t }),
+        },
+      ],
+      []
+    );
+  } else {
+    return [];
+  }
+};
+
+const getVideoServerError = ({
+  error,
+  t,
+}: {
+  error: LEServerError;
+  t: TFunction;
+}): ServerErrorItem[] => {
+  /* istanbul ignore else */
+  if (Array.isArray(error)) {
+    return Object.entries(error[0]).reduce(
+      (previous: ServerErrorItem[], [key, e]) => [
+        ...previous,
+        {
+          label: t(`event.form.labelVideo${pascalCase(key)}`),
+          message: getEventServerErrorMessage({ error: e as string[], t }),
+        },
+      ],
+      []
+    );
+  } else {
+    return [];
+  }
+};
+
+const getEventServerError = ({
+  error,
+  key,
+  t,
+  eventType,
+}: {
+  error: LEServerError;
+  eventType: string;
+  key: string;
+  t: TFunction;
+}): ServerErrorItem[] => {
+  switch (key) {
+    case 'description':
+    case 'short_description':
+      return getLocalizedServerError({ error, eventType, key, t });
+    case 'videos':
+      return getVideoServerError({ error, t });
+    default:
+      return [
+        {
+          label: getEventServerErrorLabel({
+            key,
+            eventType,
+            t,
+          }),
+          message: getEventServerErrorMessage({ error, t }),
+        },
+      ];
+  }
+};
+
+export const getEventServerErrors = ({
+  eventType,
+  result,
+  t,
+}: {
+  eventType: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  result: Record<string, any> | Record<string, any>[];
+  t: TFunction;
+}): ServerErrorItem[] =>
+  Array.isArray(result)
+    ? result.reduce(
+        (previous: ServerErrorItem[], r) => [
+          ...previous,
+          ...getEventServerErrors({
+            eventType,
+            result: r,
+            t,
+          }),
+        ],
+        []
+      )
+    : Object.entries(result).reduce(
+        (previous: ServerErrorItem[], [key, error]) => [
+          ...previous,
+          ...getEventServerError({
+            error,
+            eventType,
+            key,
+            t,
+          }),
+        ],
+        []
+      );
```

Comment:
Some comment explaining how linkedevents error are handled would be nice. This is pretty complicated function.

Also this seems like bigger error parsing module to me so having them in separate file or having those helper functions inside the exported function would help understand this. Also it would avoid passing so many parameters around again and again by using closure.

Reply:
Moved all server error parsing functions to useEventServerErrors hook to unify server error handling